plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -520,38 +545,142 @@ public boolean indexExists(String index)
+                        mappings = mappings.elements().next();
+                        if (!mappings.has("properties")) {
+                            return new IndexMetadata(outputSchema);
+                        }
                     }
-                }
 
-                JsonNode metaNode = nullSafeNode(mappings, "_meta");
+                    JsonNode metaNode = nullSafeNode(mappings, "_meta");
+                    IndexMetadata.ObjectType schema = parseType(mappings.get("properties"), nullSafeNode(metaNode, "presto"));
+                    outputSchema = merge(index, "", outputSchema, schema);
+                } while (unionSchemaIndicesForAlias && indicesIterator.hasNext());
 
-                return new IndexMetadata(parseType(mappings.get("properties"), nullSafeNode(metaNode, "presto")));
+                return new IndexMetadata(outputSchema);
             }
             catch (IOException e) {
                 throw new TrinoException(ELASTICSEARCH_INVALID_RESPONSE, e);
             }
+            catch (RuntimeException e) {
+                throw new TrinoException(ELASTICSEARCH_QUERY_FAILURE, e);
+            }
         });
     }
 
+    private IndexMetadata.ObjectType merge(String index, String parentPrefix, IndexMetadata.ObjectType schema1, IndexMetadata.ObjectType schema2)
+    {
+        if (schema2 == null || schema2.getFields().isEmpty()) {
+            return schema1;
+        }
+        if (schema1 == null || schema1.getFields().isEmpty()) {
+            return schema2;
+        }
+        List<IndexMetadata.Field> fields = merge(index, parentPrefix, schema1.getFields(), schema2.getFields());
+        return new IndexMetadata.ObjectType(fields);
+    }
+
+    private List<IndexMetadata.Field> merge(String index, String parentPrefix, List<IndexMetadata.Field> schemaFields1, List<IndexMetadata.Field> schemaFields2)
+    {
+        // Non object type fields
+        Set<IndexMetadata.Field> fields = Stream.concat(schemaFields1.stream(), schemaFields2.stream())
+                .filter(field -> !(field.getType() instanceof IndexMetadata.ObjectType))
+                .collect(Collectors.toSet());
+
+        // Object Type fields
+        Set<IndexMetadata.Field> objectTypeFields = Stream.concat(schemaFields1.stream(), schemaFields2.stream())
+                .filter(field -> field.getType() instanceof IndexMetadata.ObjectType)
+                .collect(Collectors.groupingBy(field -> field.getName()))
+                .entrySet()
+                .stream()
+                .map(fieldList -> mergeNestedFields(fieldList.getKey(), fieldList.getValue(), index, parentPrefix))
+                .collect(Collectors.toSet());
+
+        fields.addAll(objectTypeFields);
+        fields = checkAndCorrectMismatchFields(index, parentPrefix, fields);
+
+        return new ArrayList<>(fields);
+    }
+
+    private IndexMetadata.Field mergeNestedFields(String fieldName, List<IndexMetadata.Field> fields,
```

Comment:
Place each argument on a separate line when splitting across multiple lines, or keep them all on the same line.

Reply:
Done.